### PR TITLE
Fix multigene panels count

### LINKED ISSUE #5087: Genes on panel counter on variantS page gene cell repeated
**Describe the bug**
When displaying multiple genes in the same gene cell, the count of panels is for the both of them, but displayed twice.
This is slightly misleading, and maybe it wasn't even so before we merged the inheritance pattern into the cell?
Screenshot 2024-11-28 at 17 21 59
Three panels for one gene, one panel for the other:
Screenshot 2024-11-28 at 17 22 35

Thank you for the heads up @alieden!

**To Reproduce**

**Expected behavior**

**Screenshots**

**Additional context**

### PR DESCRIPTION
**Pull request by northwestwitch**
This PR adds a functionality or fixes a bug.
- On variantS page, when a variant has more than one gene, then the gene panel badge reflect the panels each gene is actually in (fix #5087)

### Before

image

### After

image

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

**How to test**:
1. how to test it, possibly with real cases/data

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] code approved by DN
- [x] tests executed by CR

### CODE REVIEW

File: scout/server/blueprints/variants/templates/variants/components.html [hunk 1]
```diff
@@ -161,94 +161,84 @@
```
Comment (northwestwitch) [inline scout/server/blueprints/variants/templates/variants/components.html:161]: I have refactored this big macro by dividing the code into 3 innner macros. It was quite messy before.

File: scout/server/blueprints/variants/templates/variants/components.html [hunk 2]
```diff
@@ -161,94 +161,84 @@
 
 {% macro gene_cell(variant, inherit_palette) %}
   <div class="align-items-center">
-    {% if variant.category == "cancer" or variant.category == "sv_cancer" %}
-      <a data-bs-toggle="tooltip" data-bs-html="true" title="
+
+    {% macro gene_tooltip(gene) %}
```
Comment (northwestwitch) [inline scout/server/blueprints/variants/templates/variants/components.html:165]: first minimacro

File: scout/server/blueprints/variants/templates/variants/components.html [hunk 3]
```diff
@@ -161,94 +161,84 @@
...
-        {{ variant.first_rep_gene.hgnc_symbol or variant.first_rep_gene.hgnc_id }}
-        {% if variant.secondary_gene %}
-          <span class="text-muted">
-            ({{ variant.second_rep_gene.hgnc_symbol or variant.second_rep_gene.hgnc_id }})
-          </span>
-        {% endif %}
+      </div>
+    {% endmacro %}
+
+    {% macro gene_link(gene) %}
```
Comment (northwestwitch) [inline scout/server/blueprints/variants/templates/variants/components.html:185]: second minimacro

File: scout/server/blueprints/variants/templates/variants/components.html [hunk 4]
```diff
@@ -161,94 +161,84 @@
...
+        data-bs-toggle="tooltip"
+        data-bs-html="true"
+        title="{{ gene_tooltip(gene) }}"
+        href="{{ url_for('genes.gene', hgnc_id=gene.hgnc_id) }}"
+      >
+        {{ gene.hgnc_symbol or gene.hgnc_id }}
       </a>
+    {% endmacro %}
+
+    {% macro panel_badge(variant, gene_id=0) %}
```
Comment (northwestwitch) [inline scout/server/blueprints/variants/templates/variants/components.html:196]: third minimacro

File: scout/server/blueprints/variants/templates/variants/components.html (hunk 3 above)
Comment (dnil) [inline scout/server/blueprints/variants/templates/variants/components.html:185]: 💯

File: scout/server/blueprints/variants/templates/variants/components.html (hunk 2 above)
Comment (dnil) [reply scout/server/blueprints/variants/templates/variants/components.html:165]: ✅

File: scout/server/blueprints/variants/templates/variants/components.html [hunk 5]
```diff
@@ -161,94 +161,84 @@
...
+        data-bs-html="true"
+        data-bs-trigger="hover click"
+        title="Overlapping gene panels"
+        data-bs-content="{% for panel in matching_panels %}
+          {{ panel.panel_name|safe }}<br>
+        {% endfor %}"
+      >{{ panel_count }}</a>
+    {% endmacro %}
+
+    {% if variant.category in ["cancer", "sv_cancer"] %}
```
Comment (dnil) [inline scout/server/blueprints/variants/templates/variants/components.html:218]: 👍

File: scout/server/blueprints/variants/templates/variants/components.html [hunk 6]
```diff
@@ -161,94 +161,84 @@
...
-            data-bs-html="true"
-            data-bs-trigger="hover click"
-            data-bs-content="{% for panel in variant.case_panels|sort(attribute='panel_name',case_sensitive=False)|rejectattr('removed') %}
-              {{ panel.panel_name|safe }}<br>
-              {% endfor %}"
-            title="Overlapping gene panels">{{panel_count}}
-          </a>
-        {% endif %}
-      </div>
+          {{ gene_link(gene) }}
```
Comment (dnil) [inline scout/server/blueprints/variants/templates/variants/components.html:233]: 👍 
Much cleaner!

**Review by dnil — APPROVED** [review]
💯

**Pull request merged**